CMakeLists.txt

```diff
@@ -65,6 +61,22 @@ if(ENABLE_TESTING)
   enable_testing()
 endif(ENABLE_TESTING)
 
+########################################################################
+#Include External projects
+########################################################################
+include(ExternalProject)
+
+set(EXTERNAL_INSTALL_LOCATION )
+
+ExternalProject_Add(spectra
+  URL https://github.com/yixuan/spectra/archive/v0.8.1.tar.gz
+  PREFIX  ${CMAKE_BINARY_DIR}/spectra
+  CONFIGURE_COMMAND "" BUILD_COMMAND "" INSTALL_COMMAND ""
+  )
+
+ExternalProject_Get_Property(spectra source_dir)
+set(SPECTRA_INCLUDE_DIR ${source_dir}/include)
+find_package(SPECTRA REQUIRED)
```

Comment:
Not sure, why this is needed if you use spectra as an external project unconditionally. Otherwise do something like:
```
find_package(SPECTRA)
if(NOT SPECTRA_FOUND)
  ExternalProject_Add(spectra ...)
 ...
endif()